token-client/src/main/java/com/sap/cloud/security/xsuaa/http/HttpHeaders.java

```diff
@@ -17,4 +18,19 @@ public HttpHeaders(List<HttpHeader> headers) {
 	public List<HttpHeader> getHeaders() {
 		return headers;
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
+			return false;
+		HttpHeaders that = (HttpHeaders) o;
+		return Objects.equals(headers, that.headers);
```

Comment:
what about the order of the headers? Is it guaranteed, that they are always added in the same order?

Reply:
For the equality of `HttpHeaders` right now the order **does** matter. In my opinion the order should not matter and I would rather change it so it doesn't!

However, for the token client this does not really matter since we control the creation of the HttpHeaders 100% and for every request type the `HttHeaders` the same.

It would still prefer to make them robust against order changes. That is a small change + test. Can I add it in this PR?

Reply:
yes, pls :-) thanks!

Reply:
Done :)